File: LeetCode/C++/leetcode_2419_Longest_Subarray_With_Maximum_Bitwise_AND.cpp
# LeetCode POTD - 2419_Longest Subarray With Maximum Bitwise AND
# Date: 2025-07-30

class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        int maxi = *max_element(nums.begin(),nums.end());
        int res = 1;
        int currLen = 0;
        for(int i=0;i<n;i++){
            if(nums[i]==maxi){
                currLen++;
            }else{
                res = max(res,currLen);
                currLen = 0;
            }

        }
        res = max(res,currLen);
        return res;
    }
};

File: LeetCode/Python/leetcode_2419_Longest_Subarray_With_Maximum_Bitwise_AND.py
# LeetCode POTD - 2419_Longest Subarray With Maximum Bitwise AND
# Date: 2025-07-30

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maxi = max(nums)
        res = 1
        curr_len = 0

        for num in nums:
            if num == maxi:
                curr_len += 1
            else:
                res = max(res, curr_len)
                curr_len = 0

        res = max(res, curr_len)
        return res
        
    
